features/step_definitions/stepdefs.js

```diff
@@ -371,6 +370,59 @@ When('the contents of the outbox is requested', async function () {
     this.response = await response.json();
 });
 
-Then('the outbox contains {int} activity', function (count) {
+When('{string} adds {string} to the Outbox', async function (actorName, activityName) {
+    if (!this.actors[actorName]) {
+        throw new Error(`Could not find Actor ${actorName}`);
+    }
+    if (!this.activities[activityName]) {
+        throw new Error(`Could not find Activity ${activityName}`);
+    }
+
+    const activity = this.activities[activityName];
+
+    // Add activity to the db
+    const activityKey = JSON.stringify([activity.id]);
+
+    if (
+        (await client('key_value').select('value').where({ key: activityKey })).length === 0
+    ) {
+        await client('key_value').insert({
+            key: activityKey,
+            value: JSON.stringify(activity)
+        })
+    }
+
+    // Add activity to the outbox
+    const outboxKey = JSON.stringify(["sites","activitypub-testing:8083","outbox"]);
+    const outbox = (
+        await client('key_value').select('value').where('key', outboxKey)
+    ).map((item) => item.value)[0] || [];
+
+    outbox.push(activity.id)
+
+    if (outbox.length === 1) {
+        await client('key_value').insert({
+            key: outboxKey,
+            value: JSON.stringify(outbox)
+        })
+    } else {
+        await client('key_value')
+        .where({ key: outboxKey })
+        .update({
+            value: JSON.stringify(outbox)
+        })
+    }
```

Comment:
Not super keen on this (exposing implementation details), but not sure of a better alternative at the mo. Open to suggestions 🙂

Reply:
Changed my mind about this and switched to a unit test